packages/react-storage/src/components/StorageManager/hooks/useUploadFiles/useUploadFiles.ts

```diff
@@ -75,43 +75,38 @@ export function useUploadFiles({
       };
 
       if (file) {
-        const {
-          file: processedFile,
-          key: processedKey,
-          ...processedRest
-        } = isFunction(processFile)
-          ? processFile({ file, key })
-          : { file, key };
-
-        onUploadStart?.({ key });
-
-        if (isResumable) {
-          const uploadTask = uploadFile({
-            ...processedRest,
-            file: processedFile,
-            fileName: path + processedKey,
-            isResumable: true,
-            level: accessLevel,
-            completeCallback: onComplete,
-            progressCallback: onProgress,
-            errorCallback: onError,
-            provider,
-          }) as unknown as UploadTask;
-          setUploadingFile({ id, uploadTask });
-        } else {
-          uploadFile({
-            ...processedRest,
-            file: processedFile,
-            fileName: path + processedKey,
-            isResumable: false,
-            level: accessLevel,
-            completeCallback: onComplete,
-            progressCallback: onProgress,
-            errorCallback: onError,
-            provider,
-          });
-          setUploadingFile({ id });
-        }
+        resolveFile({ processFile, file, key }).then(
+          ({ file, key, ...rest }) => {
```

Comment:
Don't think we need to destructure `file` any more 😅

Reply:
But since we are just directly passing the value of it below it wouldn't it just be included with `rest`?